src/common/Card/Reminder/Reminder.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect } from "react";
+import { useState } from "react";
+import { ReminderInterface } from "../../Set/interfaces";
+import * as UI from "./style";
+import PropTypes from "prop-types";
+
+/**
+ * Reminder component
+ */
+export const Reminder: React.FC<ReminderInterface> = ({
+  startTime,
+  duration,
+}) => {
+  const [currentTime, setCurrentTime] = useState(new Date());
+
+  /* Updates the reminder every 15 seconds */
+  useEffect(() => {
+    const checker = setInterval(() => {
+      setCurrentTime(new Date());
+    }, 15000);
+    return () => clearInterval(checker);
+  }, []);
+
+  /* Displays the reminder status if start time is less than 15 minutes away */
+  if (startTime.getTime() - currentTime.getTime() < 15 * 60000) {
+    const mins = Math.ceil(
+      (startTime.getTime() - currentTime.getTime()) / 60000
+    );
```

Comment:
You could probably move this calculation outside the if statement. It's not really a huge performance hit, but we're calculating the minutes twice here. You would need to remove the `* 60000` from the if statement if you move it tho